src/MinHeap/MinHeap.hpp

```diff
@@ -0,0 +1,119 @@
+#ifndef __MIN_HEAP_HPP__
+#define __MIN_HEAP_HPP__
+
+#include "inc/MinHeap/MinHeap.h"
+
+template <typename T, size_t max_size>
+bool MinHeap::MinHeap<T, max_size>::BuildMinHeap(T* unsorted_array, size_t values_count)
+{
+    if (unsorted_array == nullptr || values_count > max_size)
+    {
+        return false;
+    }
+
+    for (size_t i = 0; i < values_count; i++)
+    {
+        // We don't use memcpy here becasue we want to call copy constructor
+        m_values[i] = unsorted_array[i];
+    }
+
+    m_values_amount = values_count;
+    
+    // Run for each node except for the last to avoid underflow
+    for (size_t i = m_values_amount/2; i > 0; i--)
+    {
+        if (!MinHeapify(i))
+        {
+            return false;
+        }
+    }
+
+    if (!MinHeapify(0))
```

Comment:
I need an explanation about the underflow here

Reply:
If I were to put `>=` in the for loop, the iteration when `i` will be 0 will run, and then before the `i>=0` comparison, it would perform i--, which will be:
i = 0 - 1
but since i is unsigned, it will mean that i is UINT_MAX, which is greater than 0, and the loop will be infinite.